py3status/modules/playerctl.py

```diff
@@ -0,0 +1,360 @@
+
+    def _get_player_data(self, player):
+        data = {}
+
+        # Song attributes
+        data["album"] = player.get_album()
+        data["artist"] = player.get_artist()
+        data["title"] = player.get_title()
+        data["position"] = self._get_player_position(player)
+        self._set_data_from_metadata(player, data)
+
+        # Player attributes
+        data["player"] = player.props.player_name
+        data["loop_status"] = player.props.loop_status.value_nick
+        data["shuffle"] = player.props.shuffle
+        data["status"] = player.props.status
+        data["volume"] = int(player.props.volume * 100)
+
+        return data
+
+    def _get_player_from_index(self, index):
+        for player in self.manager.props.players:
+            if player.props.player_name == index:
+                return player
+
+        return None
+
+    def _change_player_volume(self, player, volume_factor):
+        volume_change = volume_factor * self.volume_change / 100
+        new_volume = player.props.volume + volume_change
+        try:
+            # Playerctl can't set the volume for every player
+            player.set_volume(new_volume)
+        except GLib.GError:
+            pass
+
+    def _cycle_player_loop_status(self, player):
+        new_loop_status = (player.props.loop_status + 1) % 3
+        try:
+            # Not all players support setting the loop status
+            player.set_loop_status(new_loop_status)
+        except GLib.GError:
+            pass
+
+    def _toggle_player_shuffle(self, player):
+        try:
+            # Not all players support setting the shuffle mode
+            player.set_shuffle(not player.props.shuffle)
+        except GLib.GError:
+            pass
+
+    def playerctl(self):
+        tracked_players = self.manager.props.players
+
+        players = []
+        cache_forever = True
+        for player in tracked_players:
+            player_data = self._get_player_data(player)
+
+            # Check if the player should cause the module to continuously update
+            if player_data["status"] == "Playing" and player_data["position"]:
+                cache_forever = False
+
+            # Set the color of a player
+            for key in self.thresholds_init:
+                if key in player_data:
+                    self.py3.threshold_get_color(player_data[key], key)
+
+            format_player = self.py3.safe_format(self.format_player, player_data)
+            self.py3.composite_update(format_player, {"index": player_data["player"]})
+
+            players.append(format_player)
+
+        format_player_separator = self.py3.safe_format(self.format_player_separator)
+        format_players = self.py3.composite_join(format_player_separator, players)
+
+        # Only set a cache timeout if necessary
+        if not cache_forever and self.position:
+            cached_until = self.cache_timeout
+        else:
+            cached_until = self.py3.CACHE_FOREVER
```

Comment:
I see this , I see two different conditionals for `cached_*`.

Instead of using slightly redundant `cache_forever = True` above, can you use `cached_until = self.py3.CACHE_FOREVER` instead -- and use `if self.position and .... etc` in the loop?